packages/experiments/src/components/Button/Actionable/Actionable.ts

```diff
@@ -1,11 +1,15 @@
 import { createComponent } from '../../../Foundation';
-import { ActionableStyles as styles, ActionableTokens as tokens } from './Actionable.styles';
+import { ActionableClassNames, ActionableStyles as styles, ActionableTokens as tokens } from './Actionable.styles';
 import { IActionableProps } from './Actionable.types';
 import { useButtonState as state } from '../Button.state';
 import { ActionableView } from './Actionable.view';
 
+const classNames = Object.values(ActionableClassNames);
+
 export const Actionable: React.StatelessComponent<IActionableProps> = createComponent(ActionableView, {
+  classNames,
   displayName: 'Actionable',
+  precedenceList: ['checked', 'disabled'],
```

Comment:
Is there any way to avoid the need for `precedenceList`? The `classNamesFunction` memoization didn't require any significant new API changes rather than just being able to enable/disable it. Fundamentally it seems like the approach between `createComponent` and `styled` should be able to be symmetrically implemented.

If this is absolutely required, I would want to make sure this ties into a longer-term vision of where component creation will go. This creates friction for users and people already complain that `styled` and `createComponent` are difficult to use. It may be better to exhaust non-additional API approaches (such as 'baking' slot layers into one layer) before resorting to new API requirements like these.

Reply:
This approach was just an idea that I discussed with @dzearing last week. This is by no means the final version of the memoization effort and I mostly just wanted to get a prototype out there to get some feedback. I'm more than happy to discuss whatever changes we deem necessary/consistent with our vision 😄

Reply:
Also, I think the same approach here could be used even without that precedence list. I don't know if that would impact the perf improvements we see above though. I could try that approach as well if you want.

Reply:
I just updated the PR to remove the `precedenceList` concept. From my local tests, it seems that it didn't have a significant impact on the perf improvement we were seeing. I'll update this thread when the build passes again and the new perf-test results have been posted here.

Reply:
Just to add what we talked about verbally, I wonder if one alternate idea is to bring `createComponent` in parity with `styled`.. that is, used `mergeStyleSets`/`classNamesFunction` instead of `mergeStyles`. Maybe David knows a reason offhand... but that's just one idea for pursuing a common memoization approach for both HOC helpers.